### ISSUE TITLE
Should the `fit_method` argument be maintained in `probability_epidemic()`?

### ISSUE THREAD

**Issue by Bisaloo** [epiverse-trace/superspreading#89]
Follow up from https://github.com/epiverse-trace/superspreading/pull/77#discussion_r1452542687.

At the moment `probability_epidemic()` proposes two methods to compute the probability: via `optim()` and via a grid search.

We should investigate and clarify what the cases where one is better the other is, and document it.

If one option is always better than the other (more performant & more accurate), then it should be presented as the only option to:
- avoid encouraging users to use invalid or deprecated methods
- simplify the function usage
- decrease the maintenance load

**Comment by chartgerink** [comment]
Given the state of the package + project funding at this time, I suggest we keep the existing options. 

I understand the motto of defensive programming and "prevent[ing] users from shooting themselves in the foot" – yet being overly cautious is also an option. Optimization and grid search are valid statistical methods, and without contra-indications that there is a concrete problem it seems flexible with healthy defaults.

Would it be worth closing this issue for the end of Epiverse Phase 1 or convert it into a discussion instead? @joshwlambert

**Comment by joshwlambert** [comment]
> Optimization and grid search are valid statistical methods, and without contra-indications that there is a concrete problem it seems flexible with healthy defaults.

Yes I agree. The reason for the options of numerical optimisation with `optim()` or grid search was because the original implementation used the grid search and I wanted to keep a backwards compatible option. The maintenance burden to offer more optimisation methods is relatively minimal and the current structure would allow more methods to be easily added. 

Given the considerations mentioned by @chartgerink I'll close this issue. If it is deemed relevant to change in the future it can be reopened and addressed then. 

> We should investigate and clarify what the cases where one is better the other is, and document it.

I'll review the current documentation and see if this can be improved with respect to this comment.
